CPS-0007? | Voltaire era Governance (#481)
cardano-foundation/CIPs: CPS-0004/README.md

---
CPS: 0004?
Title: "Voltaire: Cardano Governance"
Status: Open
Category: Ledger
Authors: Pi Lanningham <pi@sundaeswap.finance>
Proposed Solutions:
  - "CIP-1694: https://github.com/cardano-foundation/CIPs/pull/380"
Discussions:
  - "CIP Editors Discord, voltaire-governance channel: https://discord.gg/hdqHwSgWvG)"
Created: 2023-03-14
---

## **Abstract**

It has long been part of Cardano's vision for the "final" roadmap era to be one of Governance: allowing the community of ADA token holders to meaningfully own the decision making process by which the chain evolves.

To frame this discussion, it's important to outline a set of goals and baseline truths. Some of these will be neutral and uncontroversial, while others are in need of community discussion.

### **Acknowledgements**
<details>
<summary> Thank you to the following people who helped review this draft before it was published: </summary>

 * Adam K. Dean
 * Jared Corduan

>  * Venessa Harris

Reviewer: ```suggestion
 * Vanessa Harris
```

 - When and how to update various protocol parameters
   - These parameter changes, while limited in scope compared to a hard fork, can still have potentially existential risks on the chain
 - When and to whom to disburse funds from the treasury
   - Over the last several years, this treasury has accumulated nearly 1 billion ADA, currently worth hundreds of millions of dollars

The impact of these decisions are also often closely intertwined: development of the next hard fork may need funding from the treasury; parameters may need to be tweaked after a hard fork; etc.

Today, these decisions are primarily made by the founding entities (IOG, Cardano Foundation, and Emurgo). While these entities often consult the community when making these decisions, the final power lies in their hands, with very little that can be done beyond an uncontrolled hard fork by the stake pool operators.

For a variety of reasons, including moral, philosophical, financial, and possibly regulatory, these founding entities want to dilute their own power substantially, and share the burden of governance of the chain with the community that has arisen around it.

> IOG in particular has continued to provide skill and resources for years longer than initially intended, and is significantly over-budget and over-extended. IOG has said that they may be unable or unwilling to support development of the chain indefinitely, and so an expedient and iterative approach to governance is likely to have higher positive impact on our ability to maintain velocity as an ecosystem.

Reviewer: ```suggestion
IOG in particular has continued to provide skill and resources for years longer than initially intended. IOG has said that they may be unable or unwilling to support development of the chain indefinitely, and so an expedient and iterative approach to governance is likely to have higher positive impact on our ability to maintain velocity as an ecosystem.
```

## **Goals**

Synthesizing from the above, I posit these (hopefully uncontroversial) goals, constraints, considerations, or ground truths for any governance system. I'm only one person, though, so this list will very likely be revised and updated by collective community input as discussion unfolds.

1. A system of on-chain rules should allow decisions to be made and enacted regarding (at a minimum) the three types of changes listed above.

> 2. Every (or nearly every) ADA holder should be able to meaningfully participate in this decision making process.

Reviewer: To the 'nearly every' comment, which ADA holders are not able to participate?

Author: This is to leave space for solutions that have anti-DDOS measures like a small deposit; if you have 1 ada you can't pay the deposit, or sometimes even the transaction fees.

4. This system should not undermine it's ability to serve it's financial function, such as overburdening the chain, damaging the monetary policy, etc.
5. This system should be implementable within a reasonable time frame and budget.
6. This system should be recursively updatable; if improvements or changes to the system of governance are needed, they should be voted on and enacted via this system.
7. This system is highly dependent on the off chain user experience and community tooling, since very few users are reading raw CBOR dumps from the chain.
8. This system should be cognizant of the fact that different decisions can have different impact and risk thresholds.
9. This system cannot avoid the fact that stake pool operators of the network, collectively, always maintain control over a "pocket nuke": under sufficient discontent and misalignment with the will of governance, they can modify the code and initiate an uncontrolled hard fork to produce a separate chain aligned with their values.

Additionally, here is a more opinionated (and potentially controversial) list of potential considerations for a CIP, presented purely to foster discussion:

> 1. There is currently no known decentralized "proof of personhood" system that is well specified and easily implementable, and without such a system, "one lovelace one vote" systems are the most effective defense against sybil attacks.

Reviewer: https://go.gitcoin.co/passport is one such system

Author: I couldn't find an explanation of how it works beyond just being a standard for different centralized entities to issue "stamps"; if that is the case, who manages the "stamp issuance" for being able to participate in a vote? That's what I meant by "decentralized", but maybe I can emphasize that more.

Reviewer 2: Adding "This system cannot rely on a trusted third party. " as a core goal would help making the "proof of personhood" point less controversial (maybe? 🥲) . It makes it much clearer to the general public why 1 lovelace 1 vote is the best we can do currently.

Reviewer: From my understanding it looks to be a protocol where multiple dapps can issue 'stamps' (attestations) in your 'passport' (agreed on data format), which is basically a decentralized way to build up proof of unique humanness. Each dapp can then decide what combination of stamps sufficiently proves this. This is used as a weight on quadratic funding for gitcoin.

I squint and see PGP's web of trust.

I don't know how much of this is available or portable to Cardano.

I'm not saying 1 ADA = 1 vote is not the way to go, but it's interesting that other chains have chosen to tackle this for governance for community grants

Author: That's exactly what Atala Prism is, fwiw. And works great for specific limited domains. The mind balks at getting Cardano to agree to a specific set of stamps that qualify you though 😅 especially when we haven't even *started* laying that groundwork outside of the standard. There's not a single DID issuer with a single day of reputation yet.

Reviewer 3: :100:  :point_up:

> 7. This system is highly dependent on the off chain user experience and community tooling, since very few users are reading raw CBOR dumps from the chain.
> 8. This system should be cognizant of the fact that different decisions can have different impact and risk thresholds.
> 9. This system cannot avoid the fact that stake pool operators of the network, collectively, always maintain control over a "pocket nuke": under sufficient discontent and misalignment with the will of governance, they can modify the code and initiate an uncontrolled hard fork to produce a separate chain aligned with their values.
> 
> Additionally, here is a more opinionated (and potentially controversial) list of potential considerations for a CIP, presented purely to foster discussion:
> 
> 1. There is currently no known decentralized "proof of personhood" system that is well specified and easily implementable, and without such a system, "one lovelace one vote" systems are the most effective defense against sybil attacks.
> 2. Given the current near total control over these decisions that these founding powers have, any dilution of power that doesn't compromise the security of the chain or risk a deadlock is a worthwhile step to take.
> 3. The system loses legitimacy if participation is extremely low, or participation is severely uninformed; If a significant portion of the Cardano community has not yet had a reasonable opportunity to participate, or the majority of participation is indinstinguishable from randomness, then any decisions made during this time are illegitimate.
> 4. Not every ADA holder is interested in *directly* participating in every decision. Often an ADA holders voice can still be heard through delegation to another ADA holder.
> 5. If the system contains such a notion of delegation, the ability to withdraw, redirect, or reclaim that voice at any time is an important tool in fighting corruption.
> 6. While it may be useful to draw inspiration from existing real-world systems, it should also be understood that blockchain governance will have an entirely different set of constraints.  For example, the ADA community does not share one culture or geographic locality. Additionally, the decisions being made are not the same kinds of decisions that a world government needs to make.
> 7. The uncontrolled hard fork described above is dangerous (from a safety standpoint) and disruptive (from an economic standpoint); it will likely seriously undermine the cohesion of the Cardano community. For this reason, it should be considered a tool of absolute last resort.

Reviewer 2: ```suggestion
7. This system cannot rely on a trusted third party for "proof of personhood".  There is currently no known decentralized  system that is well specified and easily implementable, and without such a system, "one lovelace one vote" systems are the most effective defense against sybil attacks.
8. This system should be able to ensure the legitimacy of decisions. It loses legitimacy if participation is extremely low, or participation is severely uninformed; If a significant portion of the Cardano community has not yet had a reasonable opportunity to participate, or the majority of participation is indinstinguishable from randomness, then any decisions made during this time are illegitimate.

Additionally, here is a more opinionated (and potentially controversial) list of potential considerations for a CIP, presented purely to foster discussion:

1. This system is highly dependent on the off chain user experience and community tooling, since very few users are reading raw CBOR dumps from the chain.
2. This system should be cognizant of the fact that different decisions can have different impact and risk thresholds.
3. This system cannot avoid the fact that stake pool operators of the network, collectively, always maintain control over a "pocket nuke": under sufficient discontent and misalignment with the will of governance, they can modify the code and initiate an uncontrolled hard fork to produce a separate chain aligned with their values.
4. Given the current near total control over these decisions that these founding powers have, any dilution of power that doesn't compromise the security of the chain or risk a deadlock is a worthwhile step to take.
5. Not every ADA holder is interested in *directly* participating in every decision. Often an ADA holders voice can still be heard through delegation to another ADA holder.
6. If the system contains such a notion of delegation, the ability to withdraw, redirect, or reclaim that voice at any time is an important tool in fighting corruption.
7. While it may be useful to draw inspiration from existing real-world systems, it should also be understood that blockchain governance will have an entirely different set of constraints.  For example, the ADA community does not share one culture or geographic locality. Additionally, the decisions being made are not the same kinds of decisions that a world government needs to make.
8. The uncontrolled hard fork described above is dangerous (from a safety standpoint) and disruptive (from an economic standpoint); it will likely seriously undermine the cohesion of the Cardano community. For this reason, it should be considered a tool of absolute last resort.
```

Reviewer 2: Starting point 7 in Goals the points became more like "notes" which belong to the next category imo.
The point about proof of identity and the legitimacy should be stated as goals, not just notes. This is especially because these are the two points causing the most controversy, and debates should start from the same set of agreed upon goals. 

For example, solving decentralized "proof of personhood" is a long standing computer science problem and probably not going to be solved by the Cardano community (no offense). 

I like how you phrased the legitimacy problem, it gives us a good start to define "legitimacy of a decision".

Reviewer 2: Sorry again for the large diff, it seems inevitable when moving lines around, even if the content is very similar

Author: The split isn't intended to be between "goals" and "notes".

Instead, all of them are goals, constraints, considerations, etc.

And then they're split between those that I felt we're fairly uncontroversial (i.e. it's hard to imagine arguing that governance should undermine the security of the protocol), and those that still seem unsettled within the community, regardless of what my own opinions are.  For example, you and I feel that proof of personhood is likely unsolvable, but others seem unconvinced. So I'm simple being honest about how much debate there is over the point. :)

Reviewer 2: Ok I see, do you think there should be a "things we know for sure and that form the basis for CIP1694" list? Which is what I thought the "goals" section was, and why I edited it in this way.

6. What "transitionary" periods does the solution include, and what are the justifications for them? Or, conversely, what is the justification for *not* having a transitionary period?
7. What is an estimated timeline and budget for implementation of the proposed system?
8. Who is going to implement the proposed solution, and how will it be paid for?

## **Open Questions**

Below are a set of philosophical questions applicable across a broad range of solutions that are being discussed by the community:

1. Are there any "fundamental rights" of ecosystem participants that should be protected by the governance system?
2. Must any and all voting related to governance matters occur directly on the layer 1 ledger?
3. What role, if any, do the founding entities play after governance is adopted?
4. What kinds of community standards are needed outside of the scope of a specific solution at the ledger layer?
5. What kinds of incentives (or disinsentives) should we use to align behavior with our goals? Should we try to compensate actors for the time and effort they put in, or does that create corrupt and misaligned incentives of their own?

> 6. Is the ability to delegate overall more helpful or harmful? Does the risk of "popularity" based delegation (as opposed to trust / expertise based delegation) dilute the votes of others more than it increases inclusivity?

Reviewer: Does trust / expertise based delegation result in centralization that subjects Cardano to similar 'Cantillon effect' seen in fiat systems?
